### DISCUSSION TITLE
"Timed out while running tests" should say which test timed out

### DISCUSSION THREAD

**Discussion by punkpeye** [avajs/ava#3107] [Q&A]
I am running tests using `ava --verbose` and ava is exiting with:

```
  ✖ Timed out while running tests
  ✖ Exiting due to SIGINT

  47 tests passed
  2 tests skipped
```

I cannot figure out which test is the one that is timing out.

**Comment by punkpeye** [comment]
I should say that I am using ava v4.3.3, so all the related issues that I see from ~2018 I assume are irrelevant.

**Comment by punkpeye** [comment]
I should have also mentioned that... unless I SIGINT, the process just hangs indefinitely.

**Comment by novemberborn** [comment]
Are you using any loaders or require hooks, e.g. for TypeScript compilation? We've seen those get stuck and cause timeouts, but it happens at an awkward place where the tests haven't been registered yet.

**Reply by punkpeye** [reply]
I am using `tsx` loader.

**Comment by Basssiiie** [comment]
I've noticed that when there are Typescript errors, it causes a timeout. It would be nice if errors in those loaders/hooks would be logged as well, e.g...
```
✖ Typescript compilation failed with the following errors/warnings:
[...]
✖ Timed out while running tests (optionally)
```
Instead of just vaguely:
```
✖ Timed out while running tests
```
That would save a lot of frustration due to searching for why it timed out. :)

**Comment by novemberborn** [comment]
I would expect the loaders to trigger errors or write to stderr in such a way that AVA can detect it. That doesn't seem to be the case but I don't know what more AVA could do about that, it already forwards stderr and handles uncaught exceptions / rejections.
